ddtrace/internal/sampling.py

```diff
@@ -138,7 +138,7 @@ def __init__(
 
     def sample(self, span):
         # type: (Span) -> bool
-        if self.match(span) and self._sample(span):
+        if self._sample(span):
```

Comment:
Why is this change being made?

Reply:
the logic above was confusing because in order to properly know when a rule has been applied was to effectively call `match(span)` twice (once from this sample call, and a second to know if you should break from the loop).

my suggestion was to not do this match call here, and just handle "manually" in the iteration over rules.


however, I would suggest making this change in a separate PR.

Reply:
Ok I'll do it in a separate PR.

Reply:
https://github.com/DataDog/dd-trace-py/pull/3981